client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -134,274 +163,288 @@ const EventForm: React.FC<EventFormProps> = (props) => {
-        <Text>Error loading chapter</Text>
-      ) : (
-        <Heading>{dataChapter.chapter.name}</Heading>
-      )}
-      <form
-        aria-label={submitText}
-        onSubmit={handleSubmit((data) => onSubmit(data, chapterId))}
-        className={styles.form}
-      >
-        <VStack align="flex-start">
-          {fields.map((field) =>
-            field.type === 'datetime' ? (
-              <FormControl key={field.key} isRequired>
-                <DatePicker
-                  selected={field.key === 'start_at' ? startDate : endDate}
-                  showTimeSelect
-                  timeIntervals={5}
-                  onChange={onDatePickerChange(field.key)}
-                  disabled={loading}
-                  dateFormat="MMMM d, yyyy h:mm aa"
-                  customInput={
-                    <Input
-                      id={`${field.key}_trigger`}
-                      name={`${field.key}`}
-                      label={field.label}
-                      isDisabled={loading}
-                      value={
-                        field.key === 'start_at'
-                          ? startDate.toDateString()
-                          : endDate.toDateString()
-                      }
-                    />
-                  }
-                />
-              </FormControl>
-            ) : field.type === 'textarea' ? (
-              <TextArea
-                key={field.key}
-                label={field.label}
-                placeholder={field.placeholder}
-                isRequired={field.isRequired}
-                isDisabled={loading}
-                {...register(field.key)}
-              />
-            ) : (
-              <Input
-                key={field.key}
-                type={field.type}
-                label={field.label}
-                placeholder={field.placeholder}
-                isRequired={field.isRequired}
-                isDisabled={loading}
-                {...register(field.key)}
+    <form
+      aria-label={submitText}
+      onSubmit={handleSubmit(onSubmit)}
+      className={styles.form}
+    >
+      <VStack align="flex-start">
+        {!isChaptersDropdownNeeded || data ? (
+          loadingChapter ? (
+            <Text>Loading Chapter</Text>
+          ) : errorChapter || !dataChapter?.chapter ? (
+            <Text>Error loading chapter</Text>
+          ) : (
+            <Heading>{dataChapter.chapter.name}</Heading>
+          )
+        ) : (
+          <FormControl isRequired>
+            <FormLabel>Chapter</FormLabel>
+            <Select
+              {...register('chapter_id' as const, {
+                required: true,
+                valueAsNumber: true,
+              })}
+              isDisabled={loading}
+            >
+              {adminedChapters?.length &&
+                adminedChapters.map(({ id, name }) => (
+                  <option key={id} value={id}>
+                    {name}
+                  </option>
+                ))}
```

Comment:
```suggestion
              {adminedChapters.map(({ id, name }) => (
                <option key={id} value={id}>
                  {name}
                </option>
              ))}
```
Another nit, tbh. React seems to cope with an empty array, it just renders nothing.